### ISSUE TITLE
unitialized variable name wrong with no strict refs

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#10081]
Migrated from [rt.perl.org#72090](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=72090) (status was 'resolved')

Searchable as RT72090$

**Comment by p5pRT** [comment]
### From @nwc10

#### Created by @nwc10



$ \./perl \-we '$a = @&#8203;$a > 0'
Use of uninitialized value $a in array dereference at \-e line 1\.
Use of uninitialized value $a in numeric gt \(>\) at \-e line 1\.

That second warning is wrong\. It should be

Use of uninitialized value in numeric gt \(>\) at \-e line 1\.

Hash derefence has the same error in the warning for numeric gt\. Scalar
dereference does not\.

Nicholas Clark

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl 5.11.3:

Configured by nick at Tue Jan 12 11:35:18 GMT 2010.

Summary of my perl5 (revision 5 version 11 subversion 3) configuration:
  Commit id: 6f7a7d3eeb6b7da414c3496d3eb5a7c4f5127201
  Platform:
    osname=freebsd, osvers=7.0-stable, archname=i386-freebsd-thread-multi-64int
    uname='freebsd plum.flirble.org 7.0-stable freebsd 7.0-stable #7: sat jul 26 20:39:26 bst 2008 root@plum.flirble.org:usrobjusrsrcsysplum i386 '
    config_args='-Dusedevel=y -Dcc=ccache gcc -Dld=gcc -Ubincompat5005 -Uinstallusrbinperl -Dcf_email=nick@ccl4.org -Dperladmin=nick@ccl4.org -Dinc_version_list=  -Dinc_version_list_init=0 -Doptimize=-Os -Dusethreads -Duse64bitint -Accflags=-DNO_MATHOMS -Uusemymalloc -Duseperlio -Dprefix=~/Sandpit/snap5.9.x-v5.11.3-84-g6f7a7d3 -Dinstallman1dir=none -Dinstallman3dir=none -Uuserelocatableinc -de'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='ccache gcc', ccflags ='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -DNO_MATHOMS -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include',
    optimize='-Os',
    cppflags='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -DNO_MATHOMS -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.2.1 20070719  [FreeBSD]', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-pthread -Wl,-E  -fstack-protector -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib
    libs=-lm -lcrypt
    perllibs=-lm -lcrypt
    libc=, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-DPIC -fPIC', lddlflags='-shared  -L/usr/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.11.3:
    lib
    /home/nick/Sandpit/snap5.9.x-v5.11.3-84-g6f7a7d3/lib/perl5/site_perl/5.11.3/i386-freebsd-thread-multi-64int
    /home/nick/Sandpit/snap5.9.x-v5.11.3-84-g6f7a7d3/lib/perl5/site_perl/5.11.3
    /home/nick/Sandpit/snap5.9.x-v5.11.3-84-g6f7a7d3/lib/perl5/5.11.3/i386-freebsd-thread-multi-64int
    /home/nick/Sandpit/snap5.9.x-v5.11.3-84-g6f7a7d3/lib/perl5/5.11.3
    .


Environment for perl 5.11.3:
    HOME=/home/nick
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/nick/bin:/usr/flirble/admin/bin:/bin:/usr/bin:/usr/local/bin:/usr/X11R6/bin:/home/nick/bin:/usr/local/sbin:/sbin:/usr/sbin
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

Fixed by 6d1f089\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'new' to 'resolved'
